SU2_CFD/src/variable_direct_mean.cpp

```diff
@@ -837,32 +844,39 @@ bool CNSVariable::SetPrimVar(su2double eddy_visc, su2double turb_ke, CFluidModel
   check_sos   = SetSoundSpeed(FluidModel->GetSoundSpeed2());
   check_temp  = SetTemperature(FluidModel->GetTemperature());
   
+  if (check_temp) {
+    std::cout << "Encountered invalid temperature!" << std::endl;
+    std::cout << "T = " << FluidModel->GetTemperature() << std::endl;
+    std::cout << "rho = " << density << std::endl;
+    std::cout << "with E = " << GetEnergy() << ", V2 = " << Velocity2 << ", tke = " << turb_ke << std::endl;
+  }
+
   /*--- Check that the solution has a physical meaning ---*/
   
   if (check_dens || check_press || check_sos  || check_temp) {
     
-    /*--- Copy the old solution ---*/
+    // /*--- Copy the old solution ---*/
     
-    for (iVar = 0; iVar < nVar; iVar++)
-      Solution[iVar] = Solution_Old[iVar];
+    // for (iVar = 0; iVar < nVar; iVar++)
+    //   Solution[iVar] = Solution_Old[iVar];
     
-    /*--- Recompute the primitive variables ---*/
+    // /*--- Recompute the primitive variables ---*/
     
-    SetVelocity(); // Computes velocity and velocity^2
-    density = GetDensity();
-    staticEnergy = GetEnergy()-0.5*Velocity2 - turb_ke;
+    // SetVelocity(); // Computes velocity and velocity^2
+    // density = GetDensity();
+    // staticEnergy = GetEnergy()-0.5*Velocity2 - turb_ke;
     
-    /*--- Check will be moved inside fluid model plus error description strings ---*/
+    // /*--- Check will be moved inside fluid model plus error description strings ---*/
     
-    FluidModel->SetTDState_rhoe(density, staticEnergy);
+    // FluidModel->SetTDState_rhoe(density, staticEnergy);
     
-    SetDensity();
-    SetPressure(FluidModel->GetPressure());
-    SetSoundSpeed(FluidModel->GetSoundSpeed2());
-    SetTemperature(FluidModel->GetTemperature());
+    // SetDensity();
+    // SetPressure(FluidModel->GetPressure());
+    // SetSoundSpeed(FluidModel->GetSoundSpeed2());
+    // SetTemperature(FluidModel->GetTemperature());
     
     RightVol = false;
-    
+    return RightVol;
```

Comment:
What's going on here? I don't understand these changes.

Reply:
Um, this is just hackery.  The default behavior is that when a problem (i.e., a non-physical value) is encountered, it overwrites the solution at the problem point with the old solution.  To the extent that this works (i.e., allows you to keep iterating toward a steady solution), it is fine for trying to converge a steady RANS solution.  In a hybrid run, it doesn't make sense.  But, my main objective wasn't to fix the behavior, it was just to not overwrite the problem point so that I could get information about it later.  But, like I said, it's just hackery and should be taken out.

More generally, we should talk about how to handle non-physical values in hybrid runs.  My take is that the code should just stop, write a solution file so we can see what the state is, and dump some info to the screen about the failure.  One can imagine trying to recover I guess, but it is a challenging task and I don't think it is worth it right now.

Reply:
Can you make a commit with the behavior you would like?